### PR TITLE
luyr/geodesic_loss

### PR DESCRIPTION
**Pull request by luyr**
Add geodesic loss for rotation and pose loss for rotation+translation.

### CODE REVIEW

**Comment by wang-chen** [comment]
need both [functions](https://pytorch.org/docs/stable/generated/torch.nn.functional.mse_loss.html) and [modules](https://pytorch.org/docs/stable/generated/torch.nn.MSELoss.html)
